xds/src/main/java/io/grpc/xds/sds/XdsServerBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import io.grpc.BindableService;
+import io.grpc.CompressorRegistry;
+import io.grpc.DecompressorRegistry;
+import io.grpc.HandlerRegistry;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.ServerStreamTracer;
+import io.grpc.ServerTransportFilter;
+import io.grpc.netty.NettyServerBuilder;
+import io.grpc.xds.sds.internal.SdsProtocolNegotiators;
+import java.io.File;
+import java.net.InetSocketAddress;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * A wrapper around NettyServerBuilder that sets up the appropriate ProtocolNegotiator
+ * and ChannelHandlers to use SDS for TLS.
+ */
+public class XdsServerBuilder extends ServerBuilder<XdsServerBuilder> {
+
+  private final NettyServerBuilder delegate;
+
+  private XdsServerBuilder(NettyServerBuilder nettyDelegate) {
+    this.delegate = nettyDelegate;
+  }
+
+  @Override
+  public XdsServerBuilder handshakeTimeout(long timeout, TimeUnit unit) {
+    delegate.handshakeTimeout(timeout, unit);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder directExecutor() {
+    delegate.directExecutor();
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder addStreamTracerFactory(ServerStreamTracer.Factory factory) {
+    delegate.addStreamTracerFactory(factory);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder addTransportFilter(ServerTransportFilter filter) {
+    delegate.addTransportFilter(filter);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder executor(Executor executor) {
+    delegate.executor(executor);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder addService(ServerServiceDefinition service) {
+    delegate.addService(service);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder addService(BindableService bindableService) {
+    delegate.addService(bindableService);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder fallbackHandlerRegistry(@Nullable HandlerRegistry fallbackRegistry) {
+    delegate.fallbackHandlerRegistry(fallbackRegistry);
+    return this;
+  }
+
+  @Override
+  public XdsServerBuilder useTransportSecurity(File certChain, File privateKey) {
```

Comment:
how this will work with the SDS?

Reply:
Good point and ideally this should not be called when using SDS. Eventually there could be more complex cases I was thinking of - like SDS does not return any cert info so the user wants their own cert/key to be used. But for now I can change this to throw an exception or leave it as it is for now and change it to the "right thing" (whatever that happens to be) later. Comments?